### PR TITLE
Use object for Top / Left / Bottom / Right in MarginOptions

### PR DESCRIPTION
**Pull request by campersau**
Because:

1. [Upstream `PDFMargin` uses `string | number`](https://github.com/puppeteer/puppeteer/blob/3ad2e45c295083de6fc72a5041138c620615b755/packages/puppeteer-core/src/common/PDFOptions.ts#L10-L15)
2. [`PDFOptions.Width` / `PDFOptions.Height` also use `string | number` upstream](https://github.com/puppeteer/puppeteer/blob/3ad2e45c295083de6fc72a5041138c620615b755/packages/puppeteer-core/src/common/PDFOptions.ts#L125-L132)
3. `PdfOptions.Width` / `PdfOptions.Height` using `object` https://github.com/hardkoded/puppeteer-sharp/blob/e14f59da8d95536e0002efa13f451f2dc5c1c762/lib/PuppeteerSharp/PdfOptions.cs#L67-L75
4. All of these properties were already going to `ConvertPrintParameterToInches` https://github.com/hardkoded/puppeteer-sharp/blob/e14f59da8d95536e0002efa13f451f2dc5c1c762/lib/PuppeteerSharp/Cdp/CdpPage.cs#L791-L807

---

`PdfOptionsShouldBeSerializable` didn't actually test `Width` / `Height` which meant that serialisation was broken by the migration to `System.Text.Json`.
I added a simple `PrimitiveTypeConverter` to support https://github.com/hardkoded/puppeteer-sharp/issues/1001

---

`PdfOptionsShouldWorkWithMarginWithNoUnits` didn't actually test the `ConvertPrintParameterToInches` logic so I have removed it. It was just testing what `PdfOptionsShouldBeSerializable` already did.

---

Also made `MarginOptions` / `PaperFormat` records.

### CODE REVIEW

File: lib/PuppeteerSharp.Tests/PageTests/PdfTests.cs [hunk 1]
```diff
@@ -143,27 +145,5 @@ public void PdfOptionsShouldBeSerializable()
             var newPdfOptions = JsonSerializer.Deserialize<PdfOptions>(serialized);
             Assert.That(newPdfOptions, Is.EqualTo(pdfOptions));
         }
-
-        [Test]
-        public void PdfOptionsShouldWorkWithMarginWithNoUnits()
```
Comment (kblok) [inline lib/PuppeteerSharp.Tests/PageTests/PdfTests.cs:148]: This was testing something that was causing puppeteer to fail. Let's leave it so we prove that we can pass `"0"`.

Comment (campersau) [reply lib/PuppeteerSharp.Tests/PageTests/PdfTests.cs]: I added i back. I am still not sure that it actually testes the `ConvertPrintParameterToInches` code path.

https://github.com/hardkoded/puppeteer-sharp/blob/cbff8a1d6eee3f2b32fced57406d38ac941fb88a/lib/PuppeteerSharp/Cdp/CdpPage.cs#L1217-L1257

Comment (kblok) [reply lib/PuppeteerSharp.Tests/PageTests/PdfTests.cs]: That test should check [this change](https://github.com/hardkoded/puppeteer-sharp/commit/a4ce0cfab1d376bcb84c534f24a1762cd02db802#diff-b7e03018805bd7ee45bb7f4eec5eed50f785de53ec455aaa07a896af2c259a7aR1231)

Comment (kblok) [reply lib/PuppeteerSharp.Tests/PageTests/PdfTests.cs]: You're right that test is wrong.

Comment (kblok) [reply lib/PuppeteerSharp.Tests/PageTests/PdfTests.cs:148]: I turned that test into a (valid) more unit test. Let me know what you think.

**Pull request merged**